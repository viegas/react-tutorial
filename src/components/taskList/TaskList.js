import React from 'react';
import TaskItem from './taskItem/TaskItem';
import styles from './TaskList.module.css';

const TaskList = ({ list, onRemoveItem, onUpdateItem }) => {
    if (!list.length) {
        return null;
    }

    return (
        <ol className={styles.list}>
            {list.map((item) => (
                <TaskItem
                    key={item.id}
                    id={item.id}
                    text={item.text}
                    onRemoveItem={onRemoveItem}
                    onUpdateItem={onUpdateItem}
                />
            ))}
        </ol>
    );
};

export default TaskList;
