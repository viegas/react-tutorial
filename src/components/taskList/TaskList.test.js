import TaskList from './TaskList';

it('renders correctly', () => {
    const wrapper = shallow(
        <TaskList
            list={[
                { id: 1, text: 'Some text' },
                { id: 2, text: 'Some other text' },
            ]}
            onRemoveItem={() => {}}
        />
    );
    expect(wrapper.find('TaskItem')).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly with empty list', () => {
    const wrapper = shallow(<TaskList list={[]} onRemoveItem={() => {}} />);

    expect(wrapper.find('TaskItem')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
});
