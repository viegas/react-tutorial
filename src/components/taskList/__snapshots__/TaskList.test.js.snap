// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ol
  className="list"
>
  <TaskItem
    id={1}
    key="1"
    onRemoveItem={[Function]}
    text="Some text"
  />
  <TaskItem
    id={2}
    key="2"
    onRemoveItem={[Function]}
    text="Some other text"
  />
</ol>
`;

exports[`renders correctly with empty list 1`] = `""`;
